cmd/kubelet/app/options/options.go

```diff
@@ -220,7 +224,7 @@ func (f *KubeletFlags) AddFlags(fs *pflag.FlagSet) {
 	fs.StringVar(&f.CertDirectory, "cert-dir", f.CertDirectory, "The directory where the TLS certs are located. "+
 		"If --tls-cert-file and --tls-private-key-file are provided, this flag will be ignored.")
 
-	fs.StringVar(&f.CloudProvider, "cloud-provider", f.CloudProvider, "The provider for cloud services. By default, kubelet will attempt to auto-detect the cloud provider. Specify empty string for running with no cloud provider.")
+	fs.StringVar(&f.CloudProvider, "cloud-provider", f.CloudProvider, "The provider for cloud services. By default, kubelet will attempt to auto-detect the cloud provider (deprecated). Specify empty string for running with no cloud provider, this will be the default in upcoming releases.")
```

Comment:
Please add a `MarkDeprecated` e.g. `fs.MarkDeprecated("cloud-provider", "Specify empty string for running with no cloud provider, this will be the default in upcoming releases.")`

And maybe make the `StringVar` doc string start with "Deprecated", so it's super-obvious.

Reply:
@mtaufen Don't want users thinking we're deprecating the cloud-provider flag if we start the flag descrption with "DEPRECATED". Do you forsee that as a problem?

Reply:
Oh duh. I see your point, you aren't deprecating the entire flag. 
Can we at least print a deprecation warning when people use the auto-detection feature?

Reply:
Yup, I'll add that